crates/core/src/clean.rs

```diff
@@ -42,22 +42,18 @@ pub(crate) fn replace_cleaned_ranges(
     let replacement_ranges = merge_ranges(replacement_ranges);
     let mut src = src.to_string();
     for range in &replacement_ranges {
-        if let Some(replacement) = &range.1 {
-            src.replace_range(
-                range.0.start_byte as usize..range.0.end_byte as usize,
-                replacement,
-            );
-        } else {
-            src.drain(range.0.start_byte as usize..range.0.end_byte as usize);
-        }
+        src.replace_range(
+            range.range.start_byte as usize..range.range.end_byte as usize,
+            range.replacement,
+        );
     }
     Ok(Some(src))
 }
 
-pub fn get_replacement_ranges(tree: &Tree, src: &str, lang: &TargetLanguage) -> Vec<Replacement> {
+pub fn get_replacement_ranges(node: NodeWithSource, lang: &TargetLanguage) -> Vec<Replacement> {
     let mut replacement_ranges = vec![];
-    let cursor = tree.walk();
-    for n in traverse(CursorWrapper::new(cursor, src), Order::Pre) {
+    let cursor = node.node.walk();
```

Comment:
Why change this?

Reply:
@arendjr has been going through the trouble of decoupling the core crate from tree-sitter, we should support him by making use of the abstractions he added.